### ISSUE TITLE
Please Update local PubSubClient to get rid of a timeout

### ISSUE THREAD

**Issue by stefanbode** [arendst/Sonoff-Tasmota#790]
Hi Arend,

there is a change discussed and works fine for me in the PubSubClient.cpp Line 209ff

I recommend doing the change because this avoids some "Blocked loop" and speed up start time significant.

https://github.com/knolleary/pubsubclient/pull/314

```
// reads a byte into result
boolean PubSubClient::readByte(uint8_t * result) {
   uint32_t previousMillis = millis();
   while(!_client->available()) {
     uint32_t currentMillis = millis();
     yield();
     if(currentMillis - previousMillis >= ((int32_t) MQTT_SOCKET_TIMEOUT * 1000)){
       return false;
     }
   }
   *result = _client->read();
   return true;
}
```

**Comment by stefanbode** [comment]
Instead of the yield(); we can also use a delay(1); It has the same effect. The whole change is to avoid a blocking of the read procedure until the timeout(). With the delay(1) the ESP has the possibility to get more bytes from the interface and end successfully. Without it, it will never get the initially missing byte from the channel.

**Comment by arendst** [comment]
Implemented in master release v5.9.0
